src/FSharp.Stats/Testing/ConfusionMatrix.fs

```diff
@@ -0,0 +1,112 @@
+﻿namespace FSharp.Stats.Testing
+
+/// Confusion matrix for binary classification
+type BinaryConfusionMatrix<'A when 'A: equality>(
+    isPositiveLabel: 'A -> bool, 
+    actual: seq<'A>, 
+    predictions: seq<'A>
+) = 
+
+    // use mutable variables here so we do not have to iterate through the input multiple times
+    let mutable tp = 0.
+    let mutable tn = 0.
+    let mutable fp = 0.
+    let mutable fn = 0.
+
+    // get tp tn fp fn (and p/n as combinations) in one iteration
+    let _ = 
+        Seq.zip actual predictions
+        |> Seq.iter (fun (truth,pred) ->
```

Comment:
would it be a possibility to omit the mutable variables by replacing the seq.iter by a fold using an anonymous record holding with tp, tn, fp and fn as fields?

Reply:
BinaryConfusionMatrix is not only that - a record of TP/TN/FP/FN, and is used as accumulator directly - not sure how `{x with ...}` compares to increasing mutable variables though.